src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1742,6 +1742,86 @@ void Spell::EffectEnergize(SpellEffIndex effIndex)
     if (!unitTarget->IsAlive())
         return;
 
+    // Handle Secondary Potion Effects here
+    // Mad Alchemist's Potion (34440)
+    if (m_spellInfo->Id == 45051)
+    {
+        std::set<uint32> availableElixirs;
```

Comment:
This should really be in a spell script.
Also, default container for TC is std::vector.

Also, you can use initializer-list style in c++11:
```
std::vector<uint32> availableElixirs =
{
    33720, // Onslaught Elixir (28102)
    54452, // Adept's Elixir (28103)
(....)
    28489,
    28496
};